src/Decoder.cpp

```diff
@@ -135,13 +135,15 @@ int Decoder::Decode(AVPacket* pkt) {
 	int sts = VREADER_OK;
 	sts = avcodec_send_packet(decoderContext, pkt);
 	if (sts < 0 || sts == AVERROR(EAGAIN) || sts == AVERROR_EOF) {
+		av_packet_unref(pkt);
 		return sts;
 	}
 	AVFrame* decodedFrame = av_frame_alloc();
 	sts = avcodec_receive_frame(decoderContext, decodedFrame);
 
 	if (sts == AVERROR(EAGAIN) || sts == AVERROR_EOF) {
 		av_frame_free(&decodedFrame);
+		av_packet_unref(pkt);
```

Comment:
Maybe we can move unref from 146, 150 lines to 143 line, just to avoid code duplication

Reply:
@BykadorovR Fixed